apollo-api/src/main/java/com/apollographql/apollo/api/Field.java

```diff
@@ -7,6 +7,22 @@
 import java.util.List;
 import java.util.Map;
 
+/**
+ * Field is an abstraction for a field in a graphQL operation. For example, in the following graphQL query, Field
+ * represents abstraction for field 'name':
+ *
+ * <pre> {@code
+ *  {
+ *      hero {
+ *        name
+ *      }
+ *  }
+ *      }
+ * </pre>
+ *
+ * Field can refer to: <b>GraphQL Scalar Types, Objects or List</b>. For a complete list of types that a Field
+ * object can refer to see {@link Field.Type} class.
+ */
 public class Field {
```

Comment:
I feel like this class should be moved under `internal` package as it used internally by generated mappers